charmcraft/commands/build.py

```diff
@@ -440,6 +444,27 @@ def handle_dispatcher(self, linked_entrypoint):
                 relative_link = relativise(dest_hook, dispatch_path)
                 dest_hook.symlink_to(relative_link)
 
+    def handle_schemas(self):
+        """Handle fetching any defined schemas."""
+        logger.debug("Fetching any schemas from metadata.yaml")
+        metadata_yaml = self.charmdir / CHARM_METADATA
+
+        metadata = yaml.safe_load(metadata_yaml.read_text())
+
+        for rel_type in ("requires", "provides", "peers"):
+            for relation in metadata.get(rel_type, {}).values():
+                if "schema" in relation and isinstance(relation["schema"], str):
```

Comment:
I assume this is supposed to be a URL? Is there a reference for this somewhere?

If it isn't a string, should it be an error?

Reply:
Yeah, it's supposed to be a URL, and here's a reference for how the SDI library handles it:

https://github.com/canonical/serialized-data-interface/blob/main/serialized_data_interface/utils.py#L40

Here's a reference for how it's used by a charm:

https://github.com/canonical/kfp-operators/blob/master/charms/kfp-api/metadata.yaml#L23